File: 3330-ModifyTheMatrix/3330-ModifyTheMatrix.cpp
// Last updated: 8/14/2026, 2:32:49 PM
class Solution {
public:
    vector<vector<int>> modifiedMatrix(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        vector<int> colMax(n, INT_MIN);

        // First pass: compute column maxima
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                colMax[j] = max(colMax[j], matrix[i][j]);
            }
        }

        // Second pass: replace -1 with column max
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (matrix[i][j] == -1) {
                    matrix[i][j] = colMax[j];
                }
            }
        }
        return matrix;
    }
};

File: 3330-ModifyTheMatrix/3330-ModifyTheMatrix.java
// Last updated: 8/14/2026, 2:32:42 PM
class Solution {
    public int[][] modifiedMatrix(int[][] matrix) {
        int rows = matrix.length;
        int cols = matrix[0].length;

        int[] maxInCol = new int[cols];

        // Find the maximum value in each column
        for (int j = 0; j < cols; j++) {
            int max = Integer.MIN_VALUE;
            for (int i = 0; i < rows; i++) {
                max = Math.max(max, matrix[i][j]);
            }
            maxInCol[j] = max;
        }

        // Replace -1 with the maximum of its column
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                if (matrix[i][j] == -1) {
                    matrix[i][j] = maxInCol[j];
                }
            }
        }

        return matrix;
    }
}